.github/workflows/build-and-test-orchard-core.yml

```diff
@@ -10,8 +10,22 @@ on:
           are used.
 
     inputs:
+      cancel-in-progress-for-this-pr:
+        description: >
+          When set to "true", it will cancel the already running workflow for this pull request. See the concurrency
+          settings of the job for more details.
+        type: string
+        default: 'true'
+      parent-job-name:
```

Comment:
Automatically including `build-directory˙` and `build-solution-path` would make further configuration for the OSOCE and Hastlayer SDK builds (and any workflow running these for multiple solutions in the same workflow) unnecessary. So, having that if it fits in `group` would be better.

Then, while we won't (at the moment) need any further key customization, we can have a general `concurrency-key-suffix` or something parameter here, that you can use to customize the key, with the job name (added as a tip to the description) or anything else we don't think about now.

Same for `build-and-test-dotnet`.

Reply:
OK, so I slept on it. :D
I wanted to rename it to `concurrency-key-differentiator`, but that won't fit with the build-and-test-orchard-core workflow if we keep its name in the key as is (which I think is also very useful to keep like that for clarity). I'll just finish these without further changes, but hopefully they'll fix that issue about the 400 character limit.